### PR TITLE
refactor: update `blas/ext/base/ssumkbn2` to follow current project conventions

### LINKED ISSUE #1542: [RFC]: refactor `blas/ext/base/ssumkbn2` to follow current project conventions
### Description

This RFC proposes refactoring [`@stdlib/blas/ext/base/ssumkbn2`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/blas/ext/base/ssumkbn2) to follow current project conventions.

These conventions are outlined in more detail in https://github.com/stdlib-js/stdlib/issues/788. Namely, we want to migrate from C++ add-on interfaces to C add-on interfaces and then make various style and simplification changes.

Some example commits which perform most, if not all, of the requested changes include https://github.com/stdlib-js/stdlib/commit/d66d80cfcbab1da95cb5ae8cfae4a1fc4c5129d1 (saxpy),  https://github.com/stdlib-js/stdlib/commit/3c82fe7b6c0f9fd6ef02c86d6e2024d708ac8c14 (cswap), and https://github.com/stdlib-js/stdlib/commit/5b15c9675be434c0e11b1ec1e1f78d7b089abbcd (dasum).

### Related Issues

Related issues #788, #1152.

### Questions

No.

### Other

No.

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by vr-varad**
…ventions

Resolves # .

## Description

> What is the purpose of this pull request?

This pull request refactors the ```@stdlib/blas/ext/ssumkbn2``` to follow current project conventions
> ## Checklist
> * [x]  update readme.md
> * [x]  docs
> * [x]  src
> * [x]  include.gypi
> * [x]  manifest.json
> * [x]  package.json
> * [x]  lib
> * [x]  examples
> * [x]  benchmark
> * [x]  test

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #1542 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

**Comment by vr-varad** [comment]
Hello @Planeshifter and @kgryte, I'd appreciate your review of my pull request (#1749) whenever you have a moment. Thank you!

**Comment by vr-varad** [comment]
Hi @Planeshifter @Pranavchiku  and @kgryte, just checking in on my pull request (#1749). Would you mind reviewing it when you get a chance?

**Comment by vr-varad** [comment]
@Pranavchiku @Planeshifter @kgryte Any reviews updates?

**Comment by vr-varad** [comment]
@Planeshifter any updates of reviews?

File: lib/node_modules/@stdlib/blas/ext/base/ssumkbn2/README.md [hunk 1]
```diff
@@ -138,18 +129,11 @@ var v = ssumkbn2.ndarray( N, x, 2, 1 );
...
 var ssumkbn2 = require( '@stdlib/blas/ext/base/ssumkbn2' );
 
-var x;
-var i;
-
-x = new Float32Array( 10 );
-for ( i = 0; i < x.length; i++ ) {
-    x[ i ] = round( randu()*100.0 );
-}
+var x = filledarrayBy( 10, 'float64', discreteUniform( -100, 100 ) );
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/ssumkbn2/README.md]:
```suggestion
var x = filledarrayBy( 10, 'float32', discreteUniform( -100, 100 ) );
```

File: lib/node_modules/@stdlib/blas/ext/base/ssumkbn2/examples/index.js [hunk 2]
```diff
@@ -18,18 +18,11 @@
...
 var ssumkbn2 = require( './../lib' );
 
-var x;
-var i;
-
-x = new Float32Array( 10 );
-for ( i = 0; i < x.length; i++ ) {
-	x[ i ] = round( randu()*100.0 );
-}
+var x = filledarrayBy( 10, 'float64', discreteUniform( 0, 100 ) );
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/ssumkbn2/examples/index.js]:
```suggestion
var x = filledarrayBy( 10, 'float32', discreteUniform( 0, 100 ) );
```

File: lib/node_modules/@stdlib/blas/ext/base/ssumkbn2/lib/ndarray.native.js [hunk 3]
```diff
@@ -40,17 +41,14 @@ var addon = require( './ssumkbn2.native.js' );
 * var floor = require( '@stdlib/math/base/special/floor' );
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/ssumkbn2/lib/ndarray.native.js]:
```suggestion
```

File: lib/node_modules/@stdlib/blas/ext/base/ssumkbn2/lib/ndarray.native.js [hunk 4]
```diff
@@ -40,17 +41,14 @@ var addon = require( './ssumkbn2.native.js' );
...
 * // returns 5.0
 */
 function ssumkbn2( N, x, stride, offset ) {
 	var view;
-	if ( stride < 0 ) {
-		offset += (N-1) * stride;
-	}
-	view = new Float32Array( x.buffer, x.byteOffset+(x.BYTES_PER_ELEMENT*offset), x.length-offset ); // eslint-disable-line max-len
+	offset = minViewBufferIndex( N, stride, offset );
+	view = offsetView( x, offset );
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/ssumkbn2/lib/ndarray.native.js]:
```suggestion

	offset = minViewBufferIndex( N, stride, offset );

	view = offsetView( x, offset );

```

File: lib/node_modules/@stdlib/blas/ext/base/ssumkbn2/manifest.json [hunk 5]
```diff
@@ -1,40 +1,40 @@
...
+		"include": ["./include"],
+		"libraries": ["-lm"],
+		"libpath": [],
+		"dependencies": [
+		  "@stdlib/napi/export",
+		  "@stdlib/napi/argv",
+		  "@stdlib/napi/argv-int64",
+		  "@stdlib/napi/argv-strided-float32array"
+		]
+	  }
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/ssumkbn2/manifest.json]: Use spacing instead of tabs, incorrect indentation.

**Review by Pranavchiku — CHANGES_REQUESTED** [review]
Missing changes in `docs/types`, requested few changes.

**Comment by vr-varad** [comment]
@Pranavchiku I don't see any changes in docs/types to be made. Could u please review it?

**Comment by vr-varad** [comment]
@kgryte @Planeshifter @Pranavchiku Could u review it for any other issues or changes?

File: lib/node_modules/@stdlib/bench/harness/package.json [hunk 6]
```diff
@@ -28,7 +28,9 @@
     "test": "./test"
   },
   "types": "./docs/types",
-  "scripts": {},
+  "scripts": {
+    "test": "tape ./test/test.benchmark.js"
```
Comment (kgryte) [inline lib/node_modules/@stdlib/bench/harness/package.json]: Remove this. There is no need for this change and this is unrelated to this PR.

**Comment by vr-varad** [comment]
@kgryte @Pranavchiku I have made the required changes.

**Comment by vr-varad** [comment]
hey @kgryte Would you mind taking a moment to review this? Your feedback would be greatly appreciated. Thank you very much!

**Comment by vr-varad** [comment]
@kgryte I am closing this pr as there is an issue in my forked repository and will make a new pr soon.

**Pull request closed without merging**